src/lib/kafka_init.py

```diff
@@ -0,0 +1,48 @@
+import yaml
+from confluent_kafka.admin import AdminClient, NewTopic
+
+def load_topics():
+    with open('config.yaml', 'r') as file:
+        topics = yaml.safe_load(file)
+    return topics
+
+def sync_topics(admin_client, topics_config):
+    existing_topics = admin_client.list_topics(timeout=10).topics.keys()
+    topics_create = []
+    topics_to_delete = list(set(existing_topics) - set(topic_name for topic in topics_config.get("topics", []) for topic_name in topic.keys()))
```

Comment:
build the target state of each topics in a separate function for clarity. The function should return a list of topics with target state, then you can loop through that and take action.

target state should be an Enum CREATE, UPDATE, DELETE